lib/src/switch/Switch.tsx

```diff
@@ -51,73 +48,59 @@ const DxcSwitch = ({
     if (checked === undefined) {
       const isChecked = event.target.checked ?? !innerChecked;
       setInnerChecked(isChecked);
-      if (typeof onChange === "function") {
-        onChange(isChecked);
-      }
+      onChange?.(isChecked);
     } else {
-      if (typeof onChange === "function") {
-        onChange(!checked);
-      }
+      onChange?.(!checked);
     }
   };
 
   return (
     <ThemeProvider theme={colorsTheme.switch}>
-      <SwitchContainer margin={margin} size={size} onKeyDown={handleOnKeyDown}>
-        {labelPosition === "before" && hasLabel && (
+      <SwitchContainer
+        margin={margin}
+        size={size}
+        onKeyDown={handleOnKeyDown}
+        disabled={disabled}
+        onClick={!disabled ? handlerSwitchChange : undefined}
+      >
+        {labelPosition === "before" && label && (
           <LabelContainer
             id={labelId}
             labelPosition={labelPosition}
-            onClick={!disabled ? handlerSwitchChange : undefined}
             disabled={disabled}
             backgroundType={backgroundType}
-            hasLabel={hasLabel}
+            label={label}
           >
             {label} {optional && <>{translatedLabels.formFields.optionalLabel}</>}
           </LabelContainer>
         )}
-        <SwitchBase
-          labelPosition={labelPosition}
-          hasLabel={hasLabel}
-          htmlFor={labelId}
-          onClick={disabled === true ? () => {} : handlerSwitchChange}
-        >
-          <SwitchInput
+        <SwitchBase>
+          <ValueInput
```

Comment:
We need to move `ValueInput` outside the `SwitchBase` container to avoid double executing the `onChange` event. It must be a direct child of `SwitchContainer`.

Reply:
Also, it misses the `checked` property. Without it, the `value` will always be sent in the submit event.